contributors/design-proposals/daemonset-update.md

```diff
@@ -0,0 +1,196 @@
+# DaemonSet Updates
+
+**Author**: @madhusudancs, @lukaszo, @janetkuo
+
+**Status**: Proposal
+
+## Abstract
+
+A proposal for adding the update feature to `DaemonSet`.
+
+Users already can update a `DaemonSet`, which will not cause changes to its
+subsequent pods, until those pods are killed. In this proposal, we plan to add
+a "RollingUpdate" strategy which allows DaemonSet to downstream its changes to
+pods. 
+
+## Implementation 
+
+### API Object 
+
+To enable DaemonSet upgrades, the `DaemonSet` API object will have the following
+structure:
+
+```go 
+type DaemonSetUpdateStrategy struct {
+	// Type of daemon set update. Can be "RollingUpdate" or "OnDelete". 
+	// Default is OnDelete.
+	Type DaemonSetUpdateStrategyType
+
+	// Rolling update config params. Present only if DaemonSetUpdateStrategy =
+	// RollingUpdate.
+	//---
+	// TODO: Update this to follow our convention for oneOf, whatever we decide it
+	// to be. Same as DeploymentStrategy.RollingUpdate.
+	RollingUpdate *RollingUpdateDaemonSet
+}
+
+type DaemonSetUpdateStrategyType string
+
+const (
+	// Replace the old daemons by new ones using rolling update i.e replace them on each node one after the other.
+	RollingUpdateDaemonSetStrategyType DaemonSetUpdateStrategyType = "RollingUpdate"
+        
+	// Replace the old daemons only when it's killed
+	OnDeleteDaemonSetStrategyType DaemonSetUpdateStrategyType = "OnDelete"
+)
+
+// Spec to control the desired behavior of daemon set rolling update.
+type RollingUpdateDaemonSet struct {
+	// The maximum number of DaemonSet pods that can be unavailable during the
```

Comment:
Pods in a deployment are usually interchangeable (they have a load balancer in front of them).   So,  we care about the aggregate capacity of the set.   So, we limit the number unavailable to ensure sufficient aggregate capacity for the set.

Daemons often serve only their local machine.  If one is down, there is nothing to take over its load.   So, not clear that number of daemons unavailable is the main knob to control the speed of a DS rollout.